### ISSUE TITLE
Please document changes better

### ISSUE THREAD

**Issue by lespea** [pathikrit/better-files#179]
I've encountered several major changes in 3.1.0 that are not documented at all in the change log which have both runtime and compile breaking changes.  For example, you moved the entire `>` symbols to a new package (as discussed yesterday) and you used to have to returned wrapped objects when calling `autoClosed`.  There were several spots in my code that either didn't compile or compiled to unexpected things (in one case I was calling .head to get a string out and instead I was now getting a char).

Much worse, nameWithoutExtension used to chop off just the last group but now deletes everything after the first period.  I'm lucky enough that I found this early on but others probably won't be so lucky.  Now I'm concerned what other changes happened that weren't documented and I just haven't seen the behavior change yet.  I don't really want to have to go through every commit just to make sure that I have no unexpected surprised.

Sorry if I'm coming across harsh but this is pretty frustrating as an end user of a library.

**Comment by pathikrit** [comment]
You raised valid concerns. I try my best documenting them here: https://github.com/pathikrit/better-files/blob/master/CHANGES.md#v310 but it does not come with a migration guide. 

I also try to follow a semantic versioning system similar to the one used by the Scala language itself:
`MAJOR.MINOR.PATCH`

PATCH:
* No breaking API changes - only bug fixes or new APIs

MINOR:
* Minor API changes - renaming, deprecations

MAJOR:
* Major structural and API changes. Migration guide required to upgrade.

In this case you are right, maybe I should have bumped to 4.0 instead. I will try to frequent smaller incremental releases and document things better. Thank you for the feedback.

**Comment by pathikrit** [comment]
Also the symbolic operator thing happened between v2 and v3: https://github.com/pathikrit/better-files/issues/102

**Comment by lespea** [comment]
Thanks for the quick response.  Yeah I was looking at the changes document when I updated the library.  I know it's super hard staying on top of all the changes and frankly a lot of people don't even try to maintain a good changelog; so thank you for all your effort and your work on maintaining this library!

Not sure what's going on with the symbolic operator thing then... I just checked and my example from yesterday compiles in 3.0.0 without doing any extra imports.  /me shrugs.  Nbd.
